src/main/java/com/aws/greengrass/secretmanager/LocalStoreMap.java

```diff
@@ -268,28 +273,30 @@ public GetSecretValueResponse decrypt(AWSSecretResponse awsSecretResponse) throw
      * }
      */
     private void reloadSecretsFromLocalStore() {
-        SecretDocument doc;
-        try {
-             doc = secretStore.getAll();
-        } catch (SecretManagerException e) {
-            logger.atWarn().log("Cannot read secrets from the local store");
-            return;
-        }
-        if (doc == null || doc.getSecrets() == null || doc.getSecrets().isEmpty()) {
-            return;
-        }
-        for (AWSSecretResponse secretResponse : doc.getSecrets()) {
+        try (LockScope ls = LockScope.lock(localStoreLock)) {
+            SecretDocument doc;
             try {
-                decrypt(secretResponse);
-            } catch (SecretCryptoException e) {
-                logger.atWarn().kv("secretArn", secretResponse.getArn()).kv("label", secretResponse.getVersionId())
-                        .log("Unable to decrypt the secret in local store.");
-                continue;
+                doc = secretStore.getAll();
+            } catch (SecretManagerException e) {
+                logger.atWarn().log("Cannot read secrets from the local store");
+                return;
+            }
+            if (doc == null || doc.getSecrets() == null || doc.getSecrets().isEmpty()) {
+                return;
+            }
+            for (AWSSecretResponse secretResponse : doc.getSecrets()) {
+                try {
+                    decrypt(secretResponse);
```

Comment:
this is a potentially long time to be holding a lock, esp if there's a lot of secrets and the device is low powered.

Reply:
feels like complexity is creeping in. we have locks here, locks in FileSecretStore. the secrets map is concurrent....

Reply:
Absolutely. I agree with the comment. Locks in `SecretManager` are for `cache`. Locks in `LocalStoreMap` are for secrets that are read from `FileSecretStore`. I added locks wherever we're updating these objects.

Reply:
If you must have locks, please make sure to remove all lock objects and synchronized keywords. Use the cycle detecting lock factory from nucleus to get a safer lock.